app/v1/controllers/requestParsers/validators/SubmitReturnValidator.scala

```diff
@@ -0,0 +1,95 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import v1.controllers.requestParsers.validators.validations.{DateFormatValidation, DecimalValueValidation, JsonFormatValidation, NonDecimalValueValidation, PeriodKeyValidation, UniqueIDValidation, ValueFormatErrorMessages, VrnValidation}
+import v1.models.errors.MtdError
+import v1.models.request.{SubmitReturnRawData, SubmitReturnRequestBody}
+
+class SubmitReturnValidator extends Validator[SubmitReturnRawData] with ValueFormatErrorMessages{
+
+  private val validationSet = List(parameterFormatValidation, bodyValueValidator)
+
+  private def parameterFormatValidation: SubmitReturnRawData => List[List[MtdError]] = (data: SubmitReturnRawData) => {
+    List(
+      VrnValidation.validate(data.vrn),
+      JsonFormatValidation.validate[SubmitReturnRequestBody](data.body.json)
+    )
+  }
+
+  private def bodyValueValidator: SubmitReturnRawData => List[List[MtdError]] = { data =>
+    val requestBodyData = data.body.json.as[SubmitReturnRequestBody]
+
+    List(Validator.flattenErrors(List(
+      PeriodKeyValidation.validate(requestBodyData.periodKey).map(
+        _.copy(paths = Some(Seq(s"/periodKey")))
+      ),
+      DecimalValueValidation.validate(
+        amount = requestBodyData.vatDueSales,
+        path = "/vatDueSales",
+        minValue = -9999999999999.99,
+      ),
+      DecimalValueValidation.validate(
+        amount = requestBodyData.vatDueAcquisitions,
+        path = "/vatDueAcquisitions",
+        minValue = -9999999999999.99,
+      ),
+      DecimalValueValidation.validate(
+        amount = requestBodyData.totalVatDue,
+        path = "/totalVatDue",
+        minValue = -9999999999999.99,
+      ),
+      DecimalValueValidation.validate(
+        amount = requestBodyData.vatReclaimedCurrPeriod,
+        path = "/vatReclaimedCurrPeriod",
+        minValue = -9999999999999.99,
+      ),
+      DecimalValueValidation.validate(
+        amount = requestBodyData.netVatDue,
+        path = "/netVatDue",
+        maxValue = 99999999999.99,
+        message = ZERO_MINIMUM_INCLUSIVE
+      ),
+      NonDecimalValueValidation.validate(
+        amount = requestBodyData.totalValueSalesExVAT,
+        path = "/totalValueSalesExVAT"
+      ),
+      NonDecimalValueValidation.validate(
+        amount = requestBodyData.totalValuePurchasesExVAT,
+        path = "/totalValuePurchasesExVAT"
+      ),
+      NonDecimalValueValidation.validate(
+        amount = requestBodyData.	totalValueGoodsSuppliedExVAT,
+        path = "/totalValueGoodsSuppliedExVAT"
+      ),
+      NonDecimalValueValidation.validate(
+        amount = requestBodyData.totalAcquisitionsExVAT,
+        path = "/totalAcquisitionsExVAT"
+      ),
+      DateFormatValidation.validate(requestBodyData.receivedAt).map(
+        _.copy(paths = Some(Seq(s"/receivedAt")))
```

Comment:
Do you need a path here? There's only one receivedAt, it's not like ValueError which is used for multiple fields.